program/src/instructions/compressed/buy_spl.rs

```diff
@@ -193,12 +212,14 @@ pub fn process_buy_spl<'info>(
             ctx.remaining_accounts
         };
 
+    // TODO: should we really have this in the oss contract?
     // NB: TRoll hardcodes Some(100) to match
     require!(
         optional_royalty_pct == Some(100),
         TcompError::OptionalRoyaltiesNotYetEnabled
     );
 
+    // TODO: is there a world where creator accounts are not passed in first that leads to damage?
```

Comment:
The cNFT transfer will fail if the correct creator_hash is not passed in. The creator hash is of all the `Creator`s which are share, verified and address. The first two are passed in by the user as args and the third is passed in as remaining accounts. Any invalid values, e.g. the user passes in wrong accounts, wrong indices (creator shares length) will result in an invalid creator_hash.